### ISSUE TITLE
ParamsParser doesn't parse the new JSON API content type

### ISSUE THREAD

**Issue by jfelchner** [rails/rails#21168]
I'm not sure if this belongs in the MIME types gem or here so if it belongs elsewhere just let me know.

The new JSON API content type is `application/vnd.api+json`.  Which can be parsed just fine with the [current parser](https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/middleware/params_parser.rb#L17-L21) in `ParamsParser`.  Unfortunately when [the MIME type is parsed](https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/middleware/params_parser.rb#L40) it doesn't know to just use the default JSON parser.

**Comment by rodzyn** [comment]
@jfelchner Hey, can you maybe elaborate more on this, maybe preparing some examples with expected output? I was stressing my local app with `application/vnd.api+json` content type and it seemed to use default JSON parser. I've also added PR with test for this.

**Comment by radar** [comment]
I have a Rails 4.2.3 application which reproduces this issue: https://github.com/radar/rails-21168. If you comment out these lines in `config/initializers/mime_types.rb`, the test will break:

```ruby
ActionDispatch::ParamsParser::DEFAULT_PARSERS[Mime::Type.lookup('application/vnd.api+json')] = lambda do |body|
  JSON.parse(body)
end
```

(There might be a shorter way of writing that first line, but I haven't looked too deeply into it.

So what I think needs to happen is that Rails needs to be configured to automatically `JSON.parse` the body if the content type is set to `application/vnd.api+json`. You could probably update your PR to do that, @rodzyn.

**Comment by bobintornado** [comment]
Seems #20329 fixed this already. Guess could close the issue.

**Comment by sgrif** [comment]
Fixed by #20329
